cfme/fixtures/v2v_fixtures.py

```diff
@@ -0,0 +1,449 @@
+    yield rhv_hosts
+    for host in itertools.chain(rhv_hosts, vmware_hosts):
+        host.remove_credentials_rest()
+
+
+def _tag_cleanup(host_obj, tag1, tag2):
+    """
+        Clean Up Tags
+
+        Returns: Boolean True if all Tags were removed/cleaned
+        or False means all required Tags are present on host.
+    """
+
+    def extract_tag(tag):
+        # Following strip will remove extra asterisk from tag assignment
+        return tag.category.display_name.strip(" *"), tag.display_name
+
+    valid_tags = {extract_tag(tag1), extract_tag(tag2)}
+    tags = host_obj.get_tags()
+    tags_set = set(map(extract_tag, tags))
+    # we always neeed 2 tags for migration, if total is less than 2
+    # don't bother checking what tag was it, just remove it and
+    # then add all required tags via add_tags() call. or if tags on host
+    # are not subset of valid tags, we still remove them.
+    if len(tags_set) < 2 or not tags_set.issubset(valid_tags):
+        host_obj.remove_tags(tags=tags)
+        return True
+    return False
+
+
+@pytest.fixture(scope="module")
+def assign_conversion_tags(request, appliance, host_creds):
+    """Assigning tags to conversation host"""
+    tag1 = appliance.collections.categories.instantiate(
+        display_name="V2V - Transformation Host *"
+    ).collections.tags.instantiate(display_name="t")
+    if hasattr(request, "param") and request.param == "SSH":
+        transformation_method = "SSH"
+    else:
+        transformation_method = "VDDK"
+    tag2 = appliance.collections.categories.instantiate(
+        display_name="V2V - Transformation Method"
+    ).collections.tags.instantiate(display_name=transformation_method)
+    appliance.ssh_client.run_rails_command("'ConversionHost.delete_all'")
+    for host in host_creds:
+        # if _tag_cleanup() returns True, means all tags were removed
+        if _tag_cleanup(host, tag1, tag2):
+            # so we call add_tags to add only required tags
+            host.add_tags(tags=(tag1, tag2))
+            # set conversion host via rails console
+            if appliance.version >= "5.10":
+                if isinstance(host.provider, RHEVMProvider):
+                    resource_type = "Host"
+                else:
+                    resource_type = "VmOrTemplate"
+                appliance.ssh_client.run_rails_command(
+                    "'r = Host.find_by(name:{host});\
+                c_host = ConversionHost.create(name:{host},resource_id:r.id,resource_type:{type});\
+                c_host.{method}_transport_supported = true;\
+                c_host.save'".format(
+                        host=json.dumps(host.name),
+                        type=json.dumps(resource_type),
+                        method=transformation_method.lower(),
+                    )
+                )
+    yield
+    for host in host_creds:
+        host.remove_tags(tags=(tag1, tag2))
+
+
+def get_vm(request, appliance, source_provider, template, datastore="nfs"):
```

Comment:
This looks like a fixture?

`datastore` kwarg is inaccurate, its a type of datastore.

Need docblocks explaining use, assumptions, etc.